### PR TITLE
Api: Fix transforms.day() returns a format document and javadoc

### PR DESCRIPTION
**Pull request by xuzhiwen1255**

### CODE REVIEW

File: api/src/main/java/org/apache/iceberg/transforms/Transforms.java [hunk 1]
```diff
@@ -265,7 +265,7 @@ public static <T> Transform<T, Integer> day() {
    * Returns a year {@link Transform} for date or timestamp types.
```
Comment (ajantha-bhat) [inline api/src/main/java/org/apache/iceberg/transforms/Transforms.java:265]:
```suggestion
   * Returns an hour {@link Transform} for date or timestamp types.
```

File: api/src/main/java/org/apache/iceberg/transforms/Transforms.java [hunk 2]
```diff
@@ -245,7 +245,7 @@ public static <T> Transform<T, Integer> year() {
...
```
Comment (ajantha-bhat) [inline api/src/main/java/org/apache/iceberg/transforms/Transforms.java:245]:
```suggestion
   * Returns a month {@link Transform} for date or timestamp types.
```

File: api/src/main/java/org/apache/iceberg/transforms/Transforms.java [hunk 3]
```diff
@@ -255,7 +255,7 @@ public static <T> Transform<T, Integer> month() {
...
```
Comment (ajantha-bhat) [inline api/src/main/java/org/apache/iceberg/transforms/Transforms.java:255]:
```suggestion
   * Returns a day {@link Transform} for date or timestamp types.
```

File: api/src/main/java/org/apache/iceberg/transforms/Transforms.java [hunk 4]
```diff
@@ -265,7 +265,7 @@ public static <T> Transform<T, Integer> day() {
...
    *
    * @param <T> Java type passed to this transform
-   * @return a year transform
+   * @return a hour transform
```
Comment (ajantha-bhat) [inline api/src/main/java/org/apache/iceberg/transforms/Transforms.java:268]:
```suggestion
   * @return an hour transform
```

Comment (xuzhiwen1255) [reply api/src/main/java/org/apache/iceberg/transforms/Transforms.java]: Thank you for your review.

**Review by ajantha-bhat — APPROVED** [review]

**Review by nastra — APPROVED** [review]
LGTM, thanks for fixing

**Review by szehon-ho — APPROVED** [review]

**Comment by szehon-ho** [comment]
Thanks @xuzhiwen1255 for fixing, @ajantha-bhat @nastra for review

**Pull request merged**